From 2f9318b2f017e1bbfddd6c8643384de374b94afb Mon Sep 17 00:00:00 2001
From: Clemens Wolff <clemens@justamouse.com>
Date: Sat, 5 Nov 2022 01:11:05 +0100
Subject: [PATCH] Apply assorted fixes

- Remove unused id in index.html
- Move all styling to CSS
- Namespace all new CSS classes
- Make history delete button more accessible
- Remove spurious formatting changes
- Simplify history deletion implementation
---
 index.html                 |  2 +-
 mathquill4quill.css        | 10 +++++--
 mathquill4quill.js         | 56 +++++++++++++++++++-------------------
 tests/deleteButton.test.js |  5 ++--
 4 files changed, 39 insertions(+), 34 deletions(-)

diff --git a/index.html b/index.html
index a612e7e..e297ee9 100644
--- a/index.html
+++ b/index.html
@@ -49,7 +49,7 @@
           data-name="displayHistory"
           data-value="true">
       </label>
-      <label id="teste">
+      <label>
         Display delete button on formula history:
         <input
           type="checkbox"
diff --git a/mathquill4quill.css b/mathquill4quill.css
index e329507..ed2e0f3 100644
--- a/mathquill4quill.css
+++ b/mathquill4quill.css
@@ -52,6 +52,10 @@
   overflow: auto;
 }
 
+.mathquill4quill-history-container-with-delete-button {
+  width: 318px;
+}
+
 .mathquill4quill-latex-input {
   visibility: hidden !important;
   padding: 0 !important;
@@ -59,14 +63,14 @@
   width: 0 !important;
 }
 
-.close-button:after{
+.mathquill4quill-history-delete-button:after {
   display: inline-block;
   content: "\00d7";
   padding: 7px;
   cursor: pointer;
 }
 
-.inner-container{
+.mathquill4quill-history-inner-container {
   display: flex;
   justify-content: space-between;
-}
\ No newline at end of file
+}
diff --git a/mathquill4quill.js b/mathquill4quill.js
index daca465..948ddc8 100644
--- a/mathquill4quill.js
+++ b/mathquill4quill.js
@@ -1,6 +1,6 @@
 /* eslint-env browser, commonjs */
 
-window.mathquill4quill = function (dependencies) {
+window.mathquill4quill = function(dependencies) {
   dependencies = dependencies || {};
 
   const Quill = dependencies.Quill || window.Quill;
@@ -97,14 +97,13 @@ window.mathquill4quill = function (dependencies) {
       }
     }
 
-    function removeItemToHistoryList(array, index) {
+    function removeItemFromHistoryList(array, index) {
       array.splice(index, 1)
       try {
         localStorage.setItem(historyCacheKey, JSON.stringify(array));
       } catch (e) {
         // eslint-disable-line no-empty
       }
-
     }
 
     function getTooltip() {
@@ -287,17 +286,22 @@ window.mathquill4quill = function (dependencies) {
         button.setAttribute("class", "mathquill4quill-history-button");
       }
 
-      function applyInnerContainerStyles(container) {
-        container.setAttribute("class", "inner-container");
+      function applyHistoryInnerContainerStyles(container) {
+        container.setAttribute("class", "mathquill4quill-history-inner-container");
       }
 
-      function applyCloseButtonAttributes(button) {
-        button.setAttribute("class", "close-button");
+      function applyHistoryDeleteButtonStyles(button) {
+        button.setAttribute("class", "mathquill4quill-history-delete-button");
         button.setAttribute("title", "Delete");
+        button.setAttribute("role", "button");
       }
 
-      function applyHistoryContainerStyles(container) {
-        container.setAttribute("class", "mathquill4quill-history-container");
+      function applyHistoryContainerStyles(container, withDeleteButton) {
+        let className = "mathquill4quill-history-container";
+        if (withDeleteButton) {
+          className += " mathquill4quill-history-container-with-delete-button";
+        }
+        container.setAttribute("class", className);
       }
 
       function createHistoryButton(latex, mqField) {
@@ -327,33 +331,29 @@ window.mathquill4quill = function (dependencies) {
 
           historyDiv = document.createElement("div");
           let container = document.createElement("div");
-          applyHistoryContainerStyles(container);
+          applyHistoryContainerStyles(container, displayDeleteButtonOnHistory);
           let header = document.createElement("p");
           header.innerHTML = "Past formulas (max " + historySize + ")";
           historyDiv.appendChild(header);
 
-          history.forEach((element) => {
-            let innerContainer = document.createElement('div');
-            applyInnerContainerStyles(innerContainer)
+          history.forEach((element, i) => {
             const button = createHistoryButton(element, mqField);
             applyHistoryButtonStyles(button);
             if (displayDeleteButtonOnHistory) {
-              container.style.width = "318px";
+              const innerContainer = document.createElement("div");
+              applyHistoryInnerContainerStyles(innerContainer)
+              const deleteButton = document.createElement("div")
+              applyHistoryDeleteButtonStyles(deleteButton);
+              deleteButton.addEventListener("click", () => {
+                innerContainer.remove();
+                removeItemFromHistoryList(history, i);
+              });
               innerContainer.appendChild(button);
-              const closeButton = document.createElement('div')
-              applyCloseButtonAttributes(closeButton)
-              closeButton.addEventListener('click', () => {
-                const index = history.indexOf(element)
-                removeItemToHistoryList(history, index)
-                const containerToBeDeleted = document.getElementsByClassName('inner-container')
-                containerToBeDeleted[index].remove()
-              })
-              innerContainer.appendChild(closeButton);
-              container.appendChild(innerContainer)
+              innerContainer.appendChild(deleteButton);
+              container.appendChild(innerContainer);
             } else {
-              container.appendChild(button)
+              container.appendChild(button);
             }
-
           });
           historyDiv.appendChild(container);
           tooltip.appendChild(historyDiv);
@@ -376,7 +376,7 @@ window.mathquill4quill = function (dependencies) {
 
       if (
         tooltip.getBoundingClientRect().top -
-        quill.container.getBoundingClientRect().top <
+          quill.container.getBoundingClientRect().top <
         0
       ) {
         tooltip.style.top = "0px";
@@ -432,7 +432,7 @@ if (typeof module !== "undefined" && module.exports) {
 
 // for backwards compatibility with prototype-based API
 if (window.Quill) {
-  window.Quill.prototype.enableMathQuillFormulaAuthoring = function (options) {
+  window.Quill.prototype.enableMathQuillFormulaAuthoring = function(options) {
     window.mathquill4quill()(this, options);
   };
 }
diff --git a/tests/deleteButton.test.js b/tests/deleteButton.test.js
index 0d31894..6932fe9 100644
--- a/tests/deleteButton.test.js
+++ b/tests/deleteButton.test.js
@@ -18,9 +18,10 @@ module.exports = {
       .click('//a[@class="ql-action"]')
       .waitForElementVisible('//span[@class="ql-formula"]')
       .click('//button[@class="ql-formula"]')
-      .waitForElementVisible('//div[@class="mathquill4quill-history-container"]')
+      .useCss()
+      .waitForElementVisible('.mathquill4quill-history-container')
       .execute(function () {
-        document.querySelector('.close-button').click()
+        document.querySelector('.mathquill4quill-history-delete-button').click()
         return document.querySelector('.mathquill4quill-history-button')
       }, [], function (result) {
         this.assert.equal(result.value, null)
-- 
2.37.3

